### PR TITLE
Update npm-shrinkwrap.json

### PR DESCRIPTION
**Pull request by waclawjacek**
## Description

Update `npm-shrinkwrap.json` by running `npm run update-deps`.

The direct motivation to do this was to update `@woocommerce/e2e-environment` from `0.2.0` to `0.2.2` as our E2E tests are failing.

**UPDATE:** I previously thought updating `@woocommerce/e2e-environment` will resolve the issue with E2E tests. It seems this isn't the solution and what actually resolves the issue is using a different WP version which is implemented by #2450.

## Testing instructions

(Stolen from @harriswong's [similar PR](https://github.com/Automattic/woocommerce-services/pull/2059).)

1. pull this branch
2. run `npm ci`
3. create an order
4. try to print a label
5. everything should still work.

And to verify E2E tests aren't stuck at "Docker container is still being built" anymore:

1. Run `npm run docker:up`.
2. Run `npm run docker:ping`.

Output should be similar to:

```
Testing URL: http://localhost:8084

Thu Jul  8 10:03:16 CEST 2021 - Docker container is still being built
Thu Jul  8 10:03:26 CEST 2021 - Docker container is still being built
Thu Jul  8 10:03:36 CEST 2021 - Docker container is still being built
Thu Jul  8 10:03:46 CEST 2021 - Docker container had been built successfully
```

### CODE REVIEW

**Comment by waclawjacek** [comment]
I have tested:

1. Live rates and creating an order on production.
2. Creating an order and printing a shipping label on staging.

File: npm-shrinkwrap.json [hunk 1]
```diff
@@ -271,9 +271,7 @@
               }
             },
             "commander": {
-              "version": "2.20.3",
-              "resolved": "https://registry.npmjs.org/commander/-/commander-2.20.3.tgz",
```
Comment (frosso) [inline npm-shrinkwrap.json]: Weird that these entries have been removed for many packages 👀 

I did this locally with node `v10.16.0`. I updated only `@woocommerce/e2e-environment` in `package.json` to `^0.2.2`, then I ran `npm install`.
In my case I didn't get as many removals on `resolved` and `integrity`: https://github.com/Automattic/woocommerce-services/pull/2452/files

It's kinda interesting how these things happen 😅

**Review by frosso — COMMENTED** [review]
The removals of the resolved attributes looks a tiny bit sus.
When I run `npm install` with these changes on my local, the `npm-shrinkwrap.json` gets re-updated with the midding `resolved` and `integrity` attributes: https://github.com/Automattic/woocommerce-services/pull/2453/files

Do you think this might be because of slightly different nvm versions? What happens if you run `npm install` in this branch?

**Comment by waclawjacek** [comment]
> Do you think this might be because of slightly different nvm versions? What happens if you run `npm install` in this branch?

I just ran `nvm use && npm i` and committed the result to this branch. Some of the `integrity` fields are back. Thank you for catching this!

I just tried running `npm run update-deps` on my default node version (`v14.16.0`) and the command stopped with an error so I am not sure if it's possible I ran `npm run update-deps` without running `nvm use` before...

Unfortunately I'm not familiar enough with `npm` to answer this. 😞 

The branch is identical to your PR https://github.com/Automattic/woocommerce-services/pull/2453 now.

---

Actually, as I was writing this I ran `npm run update-deps` again and some of the fields are gone again - different ones though it seems. I have committed that to this branch too for comparison. Is this an intermittent thing dependent on external sources? 🤔

**Comment by frosso** [comment]
@waclawjacek thanks! 


EDIT: forgot to mention there are some file conflicts

**Comment by waclawjacek** [comment]
@frosso Oh, sorry, that's because I've committed the shrinkwrap with the fields missing so we can compare them. Resolved!

I don't know how the other changes made their way into this PR after the merge - were linting rules introduced in the meantime? Either way, I guess they don't hurt.

**Review by frosso — APPROVED** [review]
Not sure why the linting wasn't failing on those files 🤷 
Revert or keep, I'm good with both.

🚢

**Comment by waclawjacek** [comment]
The E2E tests currently fail on this branch but [they pass on `trunk`](https://github.com/Automattic/woocommerce-services/runs/3204284099). I did some local testing and learned that if I run this command on each branch:

`npm ci && npm install jest --global && npm run dist && npm run test:e2e-docker-up && npm run test:e2e-docker-ping && npm run test:e2e && npm run test:e2e-docker-down`

On `trunk`, the tests fail but all the commands are ran:

```
Test Suites: 4 failed, 4 total
Tests:       15 failed, 2 passed, 17 total
Snapshots:   0 total
Time:        94.898s
Ran all test suites.
Jest exit code: 1

> woocommerce-services@1.25.17 test:e2e-docker-down /var/www/wordpress/wp-content/plugins/woocommerce-services
> npx wc-e2e docker:down

Stopping woocommerce-services_wordpress-cli ... done
Stopping woocommerce-services_wordpress-www ... done
Stopping woocommerce-services_db            ... done
Removing woocommerce-services_wordpress-cli ... done
Removing woocommerce-services_wordpress-www ... done
Removing woocommerce-services_php           ... done
Removing woocommerce-services_db            ... done
Removing network e2e-environment_default
Removing volume e2e-environment_db
Removing volume e2e-environment_wordpress
Docker exit code: 0
```

However on this branch, the `npm run test:e2e` command fails and `npm run test:e2e-docker-down` is never ran:

```
Test Suites: 4 failed, 4 total
Tests:       11 failed, 6 passed, 17 total
Snapshots:   0 total
Time:        597.666s
Ran all test suites.
(node:35193) UnhandledPromiseRejectionWarning: Error: Protocol error (Runtime.callFunctionOn): Target closed.
    at Promise (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/Connection.js:183:56)
    at new Promise (<anonymous>)
    at CDPSession.send (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/Connection.js:182:12)
    at ExecutionContext._evaluateInternal (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/ExecutionContext.js:107:44)
    at ExecutionContext.evaluate (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/ExecutionContext.js:48:23)
    at ExecutionContext.<anonymous> (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/helper.js:112:23)
    at DOMWorld.evaluate (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/DOMWorld.js:112:20)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  -- ASYNC --
    at Frame.<anonymous> (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/helper.js:111:15)
    at Page.evaluate (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/Page.js:833:43)
    at Page.evaluate (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/helper.js:112:23)
    at _callee$ (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/@woocommerce/e2e-environment/node_modules/@wordpress/e2e-test-utils/build/@wordpress/e2e-test-utils/src/clear-local-storage.js:5:13)
    at tryCatch (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/@woocommerce/e2e-environment/node_modules/regenerator-runtime/runtime.js:63:40)
    at Generator.invoke [as _invoke] (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/@woocommerce/e2e-environment/node_modules/regenerator-runtime/runtime.js:293:22)
    at Generator.next (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/@woocommerce/e2e-environment/node_modules/regenerator-runtime/runtime.js:118:21)
    at asyncGeneratorStep (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/@woocommerce/e2e-environment/node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
    at _next (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/@woocommerce/e2e-environment/node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
    at /var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/@woocommerce/e2e-environment/node_modules/@babel/runtime/helpers/asyncToGenerator.js:32:7
  -- ASYNC --
    at Page.evaluate (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/helper.js:111:15)
    at _callee$ (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/@woocommerce/e2e-environment/node_modules/@wordpress/e2e-test-utils/build/@wordpress/e2e-test-utils/src/clear-local-storage.js:5:13)
    at tryCatch (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/@woocommerce/e2e-environment/node_modules/regenerator-runtime/runtime.js:63:40)
    at Generator.invoke [as _invoke] (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/@woocommerce/e2e-environment/node_modules/regenerator-runtime/runtime.js:293:22)
    at Generator.next (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/@woocommerce/e2e-environment/node_modules/regenerator-runtime/runtime.js:118:21)
    at asyncGeneratorStep (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/@woocommerce/e2e-environment/node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
    at _next (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/@woocommerce/e2e-environment/node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
    at /var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/@woocommerce/e2e-environment/node_modules/@babel/runtime/helpers/asyncToGenerator.js:32:7
    at new Promise (<anonymous>)
    at /var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/@woocommerce/e2e-environment/node_modules/@babel/runtime/helpers/asyncToGenerator.js:21:12
(node:35193) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 5)
(node:35193) UnhandledPromiseRejectionWarning: Error: Navigation failed because browser has disconnected!
    at CDPSession.LifecycleWatcher._eventListeners.helper.addEventListener (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/LifecycleWatcher.js:46:107)
    at CDPSession.emit (events.js:198:13)
    at CDPSession._onClosed (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/Connection.js:215:10)
    at Connection._onMessage (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/Connection.js:105:17)
  -- ASYNC --
    at Frame.<anonymous> (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/helper.js:111:15)
    at Page.goto (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/Page.js:674:49)
    at Page.goto (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/helper.js:112:23)
    at Object.openExistingOrderPage (/var/www/wordpress/wp-content/plugins/woocommerce-services/tests/e2e/utils/flows.js:81:14)
    at callback (/var/www/wordpress/wp-content/plugins/woocommerce-services/tests/e2e/specs/admin/4-refund-label.test.js:55:25)
    at callback (/var/www/wordpress/wp-content/plugins/woocommerce-services/tests/e2e/fixtures/index.js:114:10)
    at withMockedShippingLabel (/var/www/wordpress/wp-content/plugins/woocommerce-services/tests/e2e/mocks/index.js:58:8)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:35193) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 6)
(node:35193) UnhandledPromiseRejectionWarning: Error: Protocol error (Runtime.callFunctionOn): Target closed.
    at Promise (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/Connection.js:183:56)
    at new Promise (<anonymous>)
    at CDPSession.send (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/Connection.js:182:12)
    at ExecutionContext._evaluateInternal (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/ExecutionContext.js:107:44)
    at ExecutionContext.evaluateHandle (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/ExecutionContext.js:57:17)
    at ExecutionContext.<anonymous> (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/helper.js:112:23)
    at WaitTask.rerun (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/DOMWorld.js:570:65)
  -- ASYNC --
    at Frame.<anonymous> (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/helper.js:111:15)
    at Page.waitForSelector (/var/www/wordpress/wp-content/plugins/woocommerce-services/node_modules/puppeteer/lib/Page.js:1095:29)
    at clickReactButton (/var/www/wordpress/wp-content/plugins/woocommerce-services/tests/e2e/utils/index.js:71:14)
    at /var/www/wordpress/wp-content/plugins/woocommerce-services/tests/e2e/specs/admin/4-refund-label.test.js:33:10
(node:35193) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 7)
Jest exit code: 1
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! woocommerce-services@1.25.17 test:e2e: `npx wc-e2e test:e2e`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the woocommerce-services@1.25.17 test:e2e script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/waclaw/.npm/_logs/2021-08-03T11_53_40_564Z-debug.log
```

After changing the version of `@woocommerce/e2e-enviornment` in `package.json` on this branch from:

```diff
- "@woocommerce/e2e-environment": "^0.2.0",
+ "@woocommerce/e2e-environment": "0.2.0",
```

The job passes (albeit tests fail).

@frosso Should we proceed with merging this branch? It seems to me that since the E2E tests are currently finicky, this PR could be merged and the tests should be fixed in #2454.

**Comment by frosso** [comment]
> Should we proceed with merging this branch? It seems to me that since the E2E tests are currently finicky, this PR could be merged 

Yeah, I don't see it as a blocked for this PR 👍

**Comment by waclawjacek** [comment]
Left `@woocommerce/e2e-enviornment` set to `^0.2.0`.

Did a smoke test on a fresh site - everything seems fine when testing manually.

**Pull request merged**